src/background/analysis/analyze.js

```diff
@@ -144,6 +145,15 @@ function resolveBuffer(id, data) {
 async function analyze(request, userData) {
 
   const allEvidence = getAllEvidenceForRequest(request, userData);
+
+  if (await browser.cookies.getAll({url: request.reqUrl}).length != 0){
```

Comment:
It looks like you're making the same call, `await browser.cookies.getAll({url: request.reqUrl}`, in this line and then also in the one below. Better to do it once and save the result.